contracts/src/LightClient.sol

```diff
@@ -341,4 +381,78 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
             revert NoChangeRequired();
         }
     }
+
+    /// @notice check if more than threshold blocks passed since the last state update before
+    /// L1 Block Number
+    /// @param blockNumber The L1 block number
+    /// @param threshold The number of blocks updates to this contract is allowed to lag behind
+    function lagOverEscapeHatchThreshold(uint256 blockNumber, uint256 threshold)
+        public
+        view
+        returns (bool)
+    {
+        uint256 updatesCount = stateUpdateBlockNumbers.length;
+
+        // Handling Edge Cases
+        // Edgecase 1: The block is in the future or in the past before HotShot was live
+        if (blockNumber > block.number || updatesCount < 3) {
+            revert InsufficientSnapshotHistory();
+        }
+
+        uint256 prevBlock;
+        bool prevUpdateFound;
+
+        uint256 i = updatesCount - 1;
+        while (!prevUpdateFound) {
+            if (stateUpdateBlockNumbers[i] <= blockNumber) {
+                prevUpdateFound = true;
+                prevBlock = stateUpdateBlockNumbers[i];
+            }
+
+            // We don't consider the lag time for the first two updates
+            if (i < 2) {
+                break;
+            }
+            i--;
+        }
+
+        // If no snapshot is found, we don't have enough history stored to tell whether HotShot was
+        // down.
+        if (!prevUpdateFound) {
+            revert InsufficientSnapshotHistory();
+        }
+
+        return blockNumber - prevBlock > threshold;
+    }
+
+    /// @notice get the number of L1 block updates
+    function getStateUpdateBlockNumbersCount() public view returns (uint256) {
+        return stateUpdateBlockNumbers.length;
+    }
+
+    /// @notice get the HotShot commitment at the specified block height
+    /// @param hotShotBlockHeight hotShotBlockHeight
+    function getHotShotCommitment(uint256 hotShotBlockHeight)
+        public
+        view
+        returns (BN254.ScalarField hotShotCommitment)
+    {
+        uint256 commitmentsHeight = hotShotCommitments.length;
+        if (
+            hotShotCommitments[0].blockHeight > hotShotBlockHeight
+                || hotShotBlockHeight > hotShotCommitments[commitmentsHeight - 1].blockHeight
+        ) {
+            revert InvalidHotShotBlockForCommitmentCheck();
+        }
+        for (uint256 i = 1; i < commitmentsHeight; i++) {
```

Comment:
Why start at index 1? Is there a genesis commitment edge case?

Reply:
yes but has been addressed since our convo https://github.com/EspressoSystems/espresso-sequencer/commit/12b6bf256ec4136d6a2d62edaf40809818d7e6d8#diff-d85261c27bf48fe0ec00e8cc9bb3ca101d708ec68c85968531b2e2cf4f9ed292